tests/failpoints/cases/test_follower_read.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::sync::Arc;
+
+use fail;
+use std::time::Duration;
+use test_raftstore::*;
+use tikv_util::HandyRwLock;
+
+#[test]
+fn test_wait_for_apply_index() {
+    let _guard = crate::setup();
+    let mut cluster = new_server_cluster(0, 3);
+
+    // Increase the election tick to make this test case running reliably.
+    configure_for_lease_read(&mut cluster, Some(50), Some(10_000));
+    let pd_client = Arc::clone(&cluster.pd_client);
+    pd_client.disable_default_operator();
+
+    let r1 = cluster.run_conf_change();
+    cluster.must_put(b"k0", b"v0");
+    let p2 = new_peer(2, 2);
+    cluster.pd_client.must_add_peer(r1, p2.clone());
+    let p3 = new_peer(3, 3);
+    cluster.pd_client.must_add_peer(r1, p3.clone());
+    must_get_equal(&cluster.get_engine(3), b"k0", b"v0");
+
+    let region = cluster.get_region(b"k0");
+    cluster.must_transfer_leader(region.get_id(), p2.clone());
+
+    // Block all write cmd applying of Peer 3.
+    fail::cfg("on_apply_write_cmd", "sleep(5000)").unwrap();
```

Comment:
How about `pause`? It works like sleep but blocks forever until turning it off.

Reply:
The apply thread could not exit after the assertion failed (before the fail point was turned off) if we use `pause`.